sirepo/auth_db.py

```diff
@@ -67,6 +58,11 @@ def search_by(cls, **kwargs):
             with thread_lock:
                 return cls.query.filter_by(**kwargs).first()
 
+        @classmethod
+        def search_all(cls, **kwargs):
+            with thread_lock:
+                return cls.query.filter_by(**kwargs)
```

Comment:
Returning an iterator is not going to work with the thread_lock. The iteration needs to occur within the lock. There are two options: return an array or pass in an operator (code). 

I recommend redefining the function as follows:

```py
@classmethod
def search_all_for_column(cls, filter_by, column):
    with thread_lock
        return [getattr(r, column) for r in cls.query.filter_by(**filter_by)]
```

Maybe the name is wrong, but this is specifically what you need for guest_uids.